UPGRADE.md

```diff
@@ -2,6 +2,48 @@
 
 ## dev-develop
 
+### RouteBundle API
+
+The API for routes under `/admin/api/routes` has changed a bit, in order to avoid having PHP class names in the
+frontent. Instead it uses the already known `resourceKey` now.
+
+Therefore the API has changed, instead of `entityClass` it now takes the `resourceKey`, and the `entityId` attribute has
+changed to just `id`.
+
+```
+# before
+/admin/api/routes?history=true&entityClass=Sulu\Bundle\ArticleBundle\Document\ArticleDocument&entityId=c88e4b89-7e2b-4161-b5d0-c33993535140&locale=en
+
+# after
+/admin/api/routes?history=true&resourceKey=articles&id=101b58ef-d422-4122-98f9-a4009bd74bd1&locale=en
+```
+
+This `resourceKey` now has to be configured in the `sulu_route` configuration as well:
+
+```yaml
+# before
+sulu_route:
+    mappings:
+        Sulu\Bundle\ArticleBundle\Document\ArticleDocument:
+            generator: "schema"
+            options:
+                route_schema: "/articles/{object.getTitle()}"
+
+# after
+sulu_route:
+    mappings:
+        Sulu\Bundle\ArticleBundle\Document\ArticleDocument:
+            resource_key: "articles"
```

Comment:
That's maybe something we should also discuss with @chirimoya before merging it.

Reply:
I personally would, because of problems in the past with overwriting and extending entities, prefer always referencing by resource_key (so also have the resource_key in the database) and get ride completely of the entityClass in the route bundle.

Reply:
Nice idea, but that would be quite a breaking change, for which we can't even provide a migration by ourselves (because we don't know the other custom entities). We just have to decide if it is worth the breaking change.